source/ngf-mtl/impl.mm

```diff
@@ -1147,13 +1182,44 @@ ngf_error ngf_create_render_target(const ngf_render_target_info* info, ngf_rende
     NGF_NOEXCEPT {
   assert(info);
   assert(result);
+
+  bool     is_multisample       = false;
+  uint32_t ncolor_attachments   = 0;
+  uint32_t nresolve_attachments = 0;
+
+  for (uint32_t i = 0; i < info->attachment_descriptions->ndescs; ++i) {
```

Comment:
it seems strange to go through the list of attachments both here and in the `render_target` constructor.
i propose to  1) remove this code from here altogether; 2) move the `render_target` ctro code into an `initialize` method (similar to how it's done in swapchain). 3) do the check that is done here in `render_target`'s `initialize` method instead.